### PR TITLE
Switch between bookmarking endpoints based on the Tenants region.

### PR DESCRIPTION
**Pull request by kiyanwang**
This P/R updates the extension to switch between two bookmarking endpoints depending on the region of the tenancy.

- [x] Refactors to store tenant as an object as activeTenant in local storage
- [x] Updates allTenants.js to include a canadian tenancy
- [x] Updates all tenants in allTenants.js to have a new Region property, which is either EUAPAC or CA
- [x] Updates the bookmarker.js so it now accepts a tenant object and uses the region on the property on it to determine which bookmarking endpoint to use.
- [x] Updates the instructions in the README to use nvm and lock onto a version of nodejs.

Here's a screencast demonstrating it working in Chrome:
https://drive.google.com/file/d/1p10tvUSIVTl3purBO_RFDcZWRYr9MBf-/view?usp=sharing

- [x] manually test in Chrome
- [x] manually test in Firefox
- [x] manually test in MS Edge (using Chrome extension)

### CODE REVIEW

**Review by mrliptontea — COMMENTED** [review]
LGTM. I think you just need to bump the version https://github.com/talis/rl-bookmarking-extension/blob/master/package.json#L3 and it should be good to go.

**Comment by kiyanwang** [comment]
@mrliptontea I have bumped the version in `package.json`

Am just waiting for a thumb/feedback from @rgubby or @rsinger

**Review by mrliptontea — APPROVED** [review]

File: js/options.js [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
     $('#save').on('click', function() {
-        saveActiveTenant($('#tenantCode').val(), function() {
-            // Update status to let user know options were saved.
-            $('#status').html('<div class="alert alert-success">' + chromeOrBrowser().i18n.getMessage('optionsSettingsSaved') + '</div>');
-            setTimeout(function() {
-                $('#status').textContent = '';
-            }, 750);
+        getTenants(function(tenants) {
+            for (var tenantCode in tenants) {
+                if (tenantCode == $('#tenantCode').val()) {
```
Comment (rgubby) [inline js/options.js:14]: triple equals rather than double?

File: js/options.js [hunk 2]
```diff
@@ -34,21 +39,21 @@ function loadTenantList() {
         if (!activeTenant) {
             $('#optionsHelp').html('<div class="alert alert-warning">' + chromeOrBrowser().i18n.getMessage('noTenantAlert') + '</div>');
         }
-        $('#tenantCode').val(activeTenant);
+
         getTenants(function(tenants) {
             var matched = false;
             for (var tenantCode in tenants) {
-                if (tenantCode == activeTenant) {
+                if (activeTenant && tenantCode == activeTenant.code) {
```
Comment (rgubby) [inline js/options.js:46]: triple equals?

File: package.json [hunk 3]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "talis-reading-lists-bookmarker",
-  "version": "0.3.6",
+  "version": "0.3.7",
   "description": "A browser extension to bookmark resources to Talis Aspire Reading Lists",
   "devDependencies": {
     "grunt": "^1.0.1",
-    "grunt-contrib-copy": "^1.0.0"
+    "grunt-contrib-copy": "^1.0.0",
+    "manifoldjs": "^0.7.6"
```
Comment (rgubby) [inline package.json:8]: how come we needed to add this? is it a new dependency? Or one that was there, but didn't exist in the package.json file?

**Comment by rgubby** [comment]
@kiyanwang I managed to stop the video when it was going to bookmarking - you can see it's going to the right place 👍 

image

File: package.json (hunk 3 above)
Comment (kiyanwang) [reply package.json:8]: it's required (on the wiki page for packaging the ms extension) but we seem to be expecting ppl to do a `-g install` which is why I've added it as a devDependency and will update the wiki to reflect that

**Pull request merged**